SQL_2709_String_Functions.sql

```diff
@@ -0,0 +1,113 @@
+----------------------------------STRING FUNCTIONS------------------------------------------------
+
+--1. Extract the middle 3 characters from the string 'ABCDEFG'.
+SELECT SUBSTRING('ABCDEFG',3,3);
+
+--2. From a table 'Employees' with a column 'FullName', write a query to extract the first name (assuming it's always the first word before a space).
+SELECT LEFT (FullName,CHARINDEX(' ',FullName)-1) AS FirstName FROM Employees;
+--SELECT STRING_SPLIT (FullName , ' ') AS FirstName FROM Employees;;
+--3. Extract the first 5 characters from the string 'SQL Server 2022'.
+SELECT LEFT ('SQL Server 2022',5);
+
+--4. From a 'Products' table with a 'ProductCode' column, write a query to get the first 3 characters of each product code
+SELECT LEFT (ProductCode,5) from Products;
+
+--5. Extract the last 4 characters from the string 'ABCDEFGHIJKL'.
+SELECT RIGHT ('ABCDEFGHIJKL',4);
+
+--6. From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+SELECT RIGHT (OrderID,4) FROM Orders;
+
+--7. Write a query to find the length of the string 'SQL Server Functions'.
+SELECT LEN('SQL Server Functions');
+
+--8. From a 'Customers' table, find customers whose names are longer than 20 characters.
+SELECT CustName FROM Customers WHERE LEN(CustName)>20;
+
+--9. Compare the results of character count and byte count for the string 'SQL Server' with a trailing space.
+SELECT DATALENGTH('SQL Server ') AS ByteCount;  --Byte count
+SELECT LEN('SQL Server ') AS CharCount;  --Character count
+
+--10. Write a query to find the byte count of an empty string and explain the result.
+SELECT DATALENGTH('') AS ByteCount; 
+	--The  bytecount of an empty string is 0. This is because it has no characters and hence requires 0 bytes to be stored
+
+--11. Find the position of 'Server' in the string 'Microsoft SQL Server'.
+SELECT CHARINDEX('Server','Microsoft SQL Server');
+
+--12. From an 'Emails' table, write a query to extract the domain name from email addresses.
+SELECT RIGHT(EmailAddr, CHARINDEX('@',EmailAddr)) FROM Emails;

```

Comment:
Correct extraction of the domain part after @ in email addresses.

Reply:
Changed to:
SELECT RIGHT(emailid, CHARINDEX('@',REVERSE(emailid))-1) FROM Email;